components/site-search/element.js

```diff
@@ -1,20 +1,75 @@
 import { Task } from "@lit/task";
 import { LitElement, html, nothing } from "lit";
 
+import { ifDefined } from "lit/directives/if-defined.js";
+import { unsafeHTML } from "lit/directives/unsafe-html.js";
+
 import { L10nMixin } from "../../l10n/mixin.js";
 
+import "../button/element.js";
+
+// import chevronLeftIcon from "../icon/chevron-left.svg?lit";
+// import chevronRightIcon from "../icon/chevron-right.svg?lit";
+import searchIcon from "../icon/search.svg?lit";
+
 import styles from "./element.css?lit";
+
+/**
+ * @type {{[key: string]: string}}
+ */
+const localeMap = {
+  "en-us": "English (US)",
+  de: "Deutsch",
+  fr: "Français",
+  es: "Español",
+  it: "Italiano",
+  ja: "日本語",
+  ko: "한국어",
+  "pt-br": "Português (BR)",
+  ru: "Русский",
+  "zh-cn": "简体中文",
+  "zh-tw": "繁體中文",
+};
+
+/**
+ *
+ * @param {string} code
+ * @returns {string}
+ */
+function readableLocaleCode(code) {
+  return localeMap[code.toLowerCase()] || code;
+}
+
+/**
+ *
+ * @param {string} url
+ * @param {string} locale
+ * @returns {string}
+ */
+function mdnUrl2Breadcrumb(url, locale) {
+  return url
+    .replaceAll("_", " ")
+    .split("/")
+    .slice(1)
+    .filter((p) => ![locale, "docs", "Web"].includes(p))
+    .join(" / ");
+}
```

Comment:
We should put this somewhere central and reuse it in `components/search-modal/element.js`:

https://github.com/mdn/fred/blob/51fcf0ed4ff91afa5d1aba85f5eca842d55e45fd/components/search-modal/element.js#L214-L221